src/Runner.Sdk/ProcessInvoker.cs

```diff
@@ -344,6 +349,13 @@ public ProcessInvoker(ITraceWriter trace)
                 Trace.Info($"Finished process {_proc.Id} with exit code {_proc.ExitCode}, and elapsed time {_stopWatch.Elapsed}.");
             }
 
+            if (cancellationToken.IsCancellationRequested)
```

Comment:
this should go inside the using block (otherwise race condition)

Reply:
Do you mean `registration` get disposed?